natural_language_processing/word_embeddings/word_embeddings.py

```diff
@@ -0,0 +1,195 @@
+        return self._wv.similar_by_word(word)[:n]
+
+    def plot_closest_words(self, word: str, plot=True):
+        """
+        plots similar words for a given word on a 2d space
+        """
+        close_words = self.closest_words(word)
+        words = [word for word, score in close_words]
+        words.append(word.lower())
+        vectors = self._wv[words]
+        tsne = TSNE(n_components=2)
+        X_tsne = tsne.fit_transform(vectors)
+        df = pd.DataFrame(X_tsne, index=words, columns=['x', 'y'])
+        fig = plt.figure()
+        ax = fig.add_subplot(1, 1, 1)
+        ax.scatter(df['x'], df['y'])
+        for word, pos in df.iterrows():
+            ax.annotate(word, pos)
+        if plot:
+            plt.show()
+        else:
+            fig.savefig('figure.png')
+
+    def is_in_vocab(self, word: str) -> bool:
+        """
+        Checks if model's vocabulary contains the word
+        >>> w2v = WordVectors.load("word2vec.model")
+        >>> w2v.is_in_vocab("c++")
+        True
+        >>> w2v.is_in_vocab("notaword")
+        False
+        """
+        return word.lower() in self._vocab
+
+    def save(self, filename='word2vec'):
+        """
+            Saves trained model
+        """
+        full_name = filename + '.model'
+        self._model.save(full_name)
+
+    @staticmethod
+    def _parse_document(text: str) -> List[str]:
+        """
+        Parses give text document and returns list of tokens
+        >>> WordVectors._parse_document("word embeddings are the best")
+        ['word', 'embeddings', 'best']
+        """
+        tokens = word_tokenize(text)
+        splitted_tokens = []
+        for token in tokens:
+            splitted_tokens.extend(re.split('[^0-9a-zA-Z+.#]+', token))
+        parsed = [word.lower() for word in splitted_tokens
+                  if word and word not in stop_words]
+        return parsed
+
+    @classmethod
+    def train(cls, file_dir: str, size=100, window=5,
+              min_count=5, iterations=5):
+        """
+        Takes a path of directory which contains text documents
+        and trains word2vec model on the data.
+        Args:
+            file_dir - path of the directory
+            size - dimension of word embeddings
+            window - window size for word2vec algorithm
+            min_count - minimum number of occurrences for a word to be considered
+            iterations - number of iterations on a whole corpus
+        """
+        documents = []
+        full_path = os.path.join(file_dir, '*')
+        print('parsing documents')
+        for filename in glob.glob(full_path):
+            try:
+                with open(os.path.join(os.getcwd(), filename),
+                          encoding="ISO-8859-1") as f:
+                    text = f.read()
+                parsed = WordVectors._parse_document(text)
+                documents.append(parsed)
+            except IOError:
+                continue
```

Comment:
Please put lines 166 to 173 in a separate function and then do
documents = [get_document_from_file(filename) for filename in glob.glob(full_path)]